.test-infra/jenkins/SideInputTestSuite.groovy

```diff
@@ -0,0 +1,156 @@
+/*
+ * Licensed to the Apache Sextend(template)tware Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy extend(template) the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, sextend(template)tware
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import static LoadTestConfig.extendTemplate
+
+
+class SideInputTestSuite {
+    static def configurations = { LoadTestConfig template -> [
+        extendTemplate(template) {
+            title 'SideInput 2MB 100 byte records: global window'
```

Comment:
> If the side_input_size is 1000, does that mean the whole side input is 100kb in size and we are accessing it 20,000 times because that is the size of the main input?

That's right.
Would that be fine to change the size of the main input to 1 so that the side input is accessed only once? This would speed things up and maybe even allow us to set up most of your suggested test scenarios.

Reply:
> Why is the 2mb relevant in the title?

2mb is the size of the main input. I'll change the title after we decide what to do with the main input.

Reply:
@lukecwik What do you think?

Reply:
> If however the side input is a dict, to simulate accessing all the side input records, the main input and side input should have the same number of records.

I think this is not necessary. See how side input test handles dictionaries (it should be fine if you are not familiar with Python, the code is pretty simple):
https://github.com/apache/beam/blob/5e659bb80bcbf70795f6806e05a255ee72706d9f/sdks/python/apache_beam/testing/load_tests/sideinput_test.py#L127-L132

Reply:
Makes sense, thanks.